shop/src/main/java/ro/msg/learning/shop/domain/key/StockId.java

```diff
@@ -0,0 +1,47 @@
+package ro.msg.learning.shop.domain.key;
+
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Embeddable;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.UUID;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Embeddable
+@Data
+@Builder
+public class StockId implements Serializable {
+
+    @Column(name="product")
+    private UUID productId;
+    @Column(name="location")
+    private UUID locationId;
+
+@Override
+    public boolean equals(Object obj){
```

Comment:
So in lombok there is an annotation that implements equals and hashCode for you: `@EqualsAndHashCode`. Use this instead of doing the implementation please (**less code, more readable**).